lib/exercism/node.rb

```diff
@@ -0,0 +1,56 @@
+# {:path=>"README.txt",
+#    :mode=>"100644",
+#    :type=>"blob",
+#    :sha=>"e69de29bb2d1d6434b8b29ae775ad8c2e48c5391",
+#    :size=>0,
+#    :url=>""}
+
+# CONVERTED TO
+
+# "[{\"id\":\"README.txt\",\"parent\":\"#\",\"icon\":\"file\",
+#    \"data\":{\"sha\":\"e69de29bb2d1d6434b8b29ae775ad8c2e48c5391\",
+#    \"type\":\"file\"}}]"
+
+class Node
+  attr_reader :node
+
+  def initialize(node)
+    @node = node
+  end
+
+  def id
+    path_names.last
+  end
+  
+  def parent
+    if root_node?
+      "#"
+    else
+      path_names.pop
+      path_names.join("/")
+    end
+  end
+
+  def icon
+    node.type.eql?('tree') ? '' : 'file'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.